packages/core/src/routes/session.ts

```diff
@@ -108,40 +111,27 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
     '/session/register',
     koaGuard({
       body: object({
-        username: string().min(3),
-        password: string().min(6),
+        username: string().min(3).optional(),
+        password: string().min(6).optional(),
+        email: string().optional(),
+        code: string().optional(),
       }),
     }),
     async (ctx, next) => {
-      const { username, password } = ctx.guard.body;
-
-      if (await hasUser(username)) {
-        throw new RequestError('user.username_exists');
+      const interaction = await provider.interactionDetails(ctx.req, ctx.res);
+      const { jti } = interaction;
+      const { username, password, email, code } = ctx.guard.body;
+
+      if (email && !code) {
+        await sendRegisterWithEmailPasscode(ctx, jti, email);
```

Comment:
will it be better to have "send passcode" in a separate route?

Reply:
In previous discussions on Linear, the conclusion was no seperate route.

Both are good for me, let's hold it for now and refactor if necessary after current core projects are closed? It is wraped into a single function and is easy to move.

Reply:
tracking in LOG-1482

Reply:
np